File: Javascript/Coin_Change.js
// amount = total amount, coins = array of coin
function coinChange(amount, coins){
    let storedAmounts = Array(amount + 1).fill(0);
    storedAmounts[0] = 1;
    for(let coin of coins){
        for(let currentAmount of storedAmounts.keys()){
            if(currentAmount >= coin){
                let remainder = currentAmount - coin;
                storedAmounts[currentAmount] += storedAmounts[remainder];
            }
        }
    }
    return storedAmounts.pop();
}

console.log(coinChange(12, [1, 2, 5]));

File: Python/Coin_Change.py
#Question: Given a set of coins to chose from, and a target amount, find the number of different combinations you can have to make that amount 
#Solution: Dynamic Programming solution (DP)

# amount = total amount, coins = array of coin
def coinChange(amount, coins):
    store = [0] * (amount + 1)
    store[0] = 1
    for coin in coins:
        for j in range(len(store)):
            if(j >= coin):
                store[j] += store[j-coin]
    print(store)


print(coinChange(3, [1, 2]))